File: src/test/baekjoon/java/greedy/StringCurrency.java
package test.baekjoon.java.greedy;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.StringTokenizer;
import java.util.stream.Stream;

import static java.util.stream.Collectors.joining;

public class StringCurrency {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        StringTokenizer st = new StringTokenizer(br.readLine());
        int N = Integer.parseInt(st.nextToken());
        int X = Integer.parseInt(st.nextToken());

        if (N > X || (N * 26) < X) {
            System.out.println("!");
        } else {
            char[] answer = new char[N];
            Arrays.fill(answer, 'A');

            X -= N;
            int idx = N - 1;

            while (X > 0) {
                if (X >= 25) {
                    answer[idx--] = 'Z';
                    X -= 25;
                } else {
                    answer[idx] = (char) (X + 65);
                    break;
                }
            }

            System.out.println(Stream.of(answer).map(String::valueOf).collect(joining()));
        }
    }
}

File: src/test/baekjoon/python/greedy/StringCurrency.py
N, X = map(int, input().split())

if (N * 26) < X or N > X:
    print('!')
else:
    answer = ['A'] * N
    X -= N
    idx = N - 1

    while X > 0:
        if X >= 25:  # 초기 A 세팅. A -> 다른 알파벳으로 바꾸는 비용 (알파벳 숫자 - 1)
            answer[idx] = 'Z'
            X -= 25
            idx -= 1
        else:
            answer[idx] = chr(X + 65)
            break

    print("".join(answer))
